### PR TITLE
Single Lao Activity

### PR DESCRIPTION
**Pull request by Warhedgehog**
Alright big one quite sorry about it, though some of the changes come from #1373 and #1384 which should be merged before this.

This adopts a single-activity architecture for everything LAO related. In practice this means merging Social Media and Digital Cash into the LaoDetailActivity.
Consequently this introduces several viewModels for this new activity. One is generic, with Lao related methods, and roughly each feature has one as well.

Unfortunately this transition comes at a cost. 

1. The scanning shared by the Home and the Lao is limit. `HomeViewModel` now has to implement a few useless methods. This poses no practical problems but architectural ones. Therefore it might make sense to do separate fragments in the future (though this will induce a bit of code duplication), especially if new features need the scanner.
2. There is no good way to only display the transaction history button on the toolbar only when a digital cash fragment is selected. We have several options : 
    - The way I do it now is it is always displayed
    - Alternative could be having 2 identical toolbars in the layout (except that one has the icon) and switch visibilty on the 2 based on what fragment is displayed
    - Not have that icon in the toolbar but in the top right corner of the digital cash home screen
  
Also I "broke" the witness message list. Now nothing is displayed on them. I did not fix it because it needs to redone from scratch anyway and the PR is big enough I believe

### CODE REVIEW

**Comment by pierluca** [comment]
Wow, it looks like you have quit a few merge conflicts here @Warhedgehog

**Comment by jbsv** [comment]
@Warhedgehog is there a 'good' way to approach this  bigg review ? any suggestion ?

**Comment by Warhedgehog** [comment]
@jbsv, you should be able to review commit by commit, as I did my best to organize things in that regard. Also dismiss no need to care about any change in the `LaoDetailActivity`, `SocialMediaActivity`,  and `LaoDetailViewModel` as they got deleted in the end. Also while there are a lot of changed files, most of them have just been moved to reflect the new architecture (in retrospect I should have done that in a separate PR, sorry). Moreover, for most fragment the change comes down to changing what view models they use and which activity they refer to, which is quite a few lines of code (also with imports).

So, I think the best strategy is to review all commits separately except the first and then review the following files: `LaoActivity`, all view models (except `LaoDetailViewModel`) and the `QrScannerFragment`.  The first commit is most of the `LaoActivity` and its view model but it got fixed/adapted along the way so it makes more sense to review the final version.

Hope this helps and don't hesitate if you have a question

**Review by jbsv — APPROVED** [review]
Very nice job. I just wonder how much this rather big change has been tested. But even if not completely tested, I fill that the cleaner code will help catch up little bugs and is a better basis for students to start with.

File: fe2-android/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -182,7 +182,7 @@
   <!-- Digital Cash -->
   <string name="digital_cash_send">Send</string>
   <string name="digital_cash_issue">Issue</string>
-  <string name="digital_cash_history">History</string>
+  <string name="digital_cash_history">Transaction History</string>
```
[the remark this answers is not in the document]
Comment (Warhedgehog) [reply fe2-android/app/src/main/res/values/strings.xml:185]: I'd say the current version is better. According to [Cornell Law School,](https://www.law.cornell.edu/definitions/uscode.php?width=840&height=800&iframe=true&def_id=21-USC-915630766-1306475316&term_occur=1&term_src=title:21:chapter:9:subchapter:V:part:H:section:360eee#:~:text=The%20term%20“transaction%20history”%20means,the%20manufacturer%20of%20the%20product.)

> The term “transaction history” means a statement in paper or electronic form, including the transaction information for each prior transaction (...).

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoViewModel.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+  protected void updateRole() {
+    Role currentRole = determineRole();
+    if (role.getValue() != currentRole) {
+      this.role.setValue(currentRole);
+    }
+  }
+
+  /**
+   * This function should be used to add disposable object generated from subscription to sent
```
[the remark this answers is not in the document]
Comment (Warhedgehog) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoViewModel.java:173]: Sent, as it is the adjective :)

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoActivity.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+    // Update the user's role in the drawer header when it changes
+    viewModel.getRole().observe(this, this::setupHeaderRole);
+  }
+
+  private void observeToolBar() {
+    // Listen to click on left icon of toolbar
+    binding.laoAppBar.setNavigationOnClickListener(
+        view -> {
+          if (Boolean.TRUE.equals(viewModel.isTab().getValue())) {
```
[the remark this answers is not in the document]
Comment (Warhedgehog) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/LaoActivity.java]: We cannot as isTab() is not a boolean, it is a mutableLiveData wrapping a boolean. It must be so, in order to observe the variable

**Pull request merged**